src/structures/APIMessage.js

```diff
@@ -98,11 +110,23 @@ class APIMessage {
     return content;
   }
 
+  /**
+   * Resolves both data and files.
+   * @returns {APIMessage}
+   */
+  async resolve() {
```

Comment:
I'm thinking that `resolveData` and `resolveFiles` should return their respective thing instead of returning the instance.
It makes more sense in my opinion and a return of `Promise<this>` is very ugly too.
Would also mean there would be no  need for `resolve` here.

Reply:
`resolve` is fine to go, but in practice (see both send methods) returning this and Promise<this> is more useful than the data... The methods should just mutate the cache and return itself